### DISCUSSION TITLE
Deskew image

### DISCUSSION THREAD

**Discussion by user1996-tech** [ImageMagick/ImageMagick#2267] [Help] [answered]
I'm trying to deskew a binary image of a receipt so that I cant run OCR it the text more accurately. 
Does any one know a solution to this problem ?

Any help would be much appreciated.
https://user-images.githubusercontent.com/66292522/87509589-36dc2a00-c6b5-11ea-9286-6e4993a454c3.jpg

**Comment by fmw42** [comment, marked as answer]
Pick the 4 corner coordinates of the paper strip in your image and define 4 corresponding corners in your desired output image having about the same width and height as the paper strip. Then do a perspective warp. See https://imagemagick.org/Usage/distorts/#perspective

What is your IM version and platform/OS?

**Reply by user1996-tech** [reply]
Ok thanks, will give it a shot. 
Version: ImageMagick 6.9.10-23
OS: Ubuntu 20

**Reply by user1996-tech** [reply]
I've tried that approach and it works, thanks for that. Is there also a way to automatically draw a rectangle around the white object and not manually select pixels ?

**Reply by fmw42** [reply]
See minimum rotated bounding box at https://imagemagick.org/script/convex-hull.php#box, but you will need to do something to remove the noise, such as -connected-components processing at https://imagemagick.org/script/connected-components.php
